### PR TITLE
CON-200 Prevent users from having multiple roles

### PR DESCRIPTION
**Pull request by cop1fab**
#### Description
This Pull request checks if a user was already assigned a role and doesn't allow for a duplicate in the role's table. 

#### Context of the problem to be solved. 

The user was allowed to be assigned two roles, now a user should only have one role.  

#### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/mrm_api.git`
2. Setup project according to Readme.md
3. checkout to branch `story/CON-200-validating-duplicates`
4. run the following query to create a user role. Make sure this user already exists and has at least one role assigned to him/her
```
mutation{
  createUserRole(userId: (user-role), roleId: 2){
    userRole{
      id
      roles{
        id
      }
    }
  }
}``

#### Type of Change
- [x] New feature (non-breaking change which adds functionality

#### How has this been tested
- [x] Unit tests
- [x] Integration tests

#### Checklist
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] Existing unit tests pass locally with my changes
- [x] Implementation works according to expectations



#### JIRA
[CON-200](https://andela-apprenticeship.atlassian.net/browse/CON-200)

### CODE REVIEW

File: api/user/schema.py [hunk 1]
```diff
@@ -212,11 +212,11 @@ def mutate(self, info, **kwargs):
         location_id = kwargs['location_id']
         query_user = User.get_query(info)
         user = query_user.filter(
-          UserModel.state == "active", UserModel.email == email).first()
+            UserModel.state == "active", UserModel.email == email).first()
```
Comment (IssaIan) [inline api/user/schema.py]: You introduced new changes which do not have any effect. This is also on line 219. Please check on this.

Comment (cop1fab) [reply api/user/schema.py]: Flake8 suggested those changes.

Comment (IssaIan) [reply api/user/schema.py]: Okay then.

Comment (rajeman) [reply api/user/schema.py]: @cop1fab Kindly invalidate this change. It is not part of the task

File: api/user/schema.py [hunk 2]
```diff
@@ -212,11 +212,11 @@ def mutate(self, info, **kwargs):
...
         if not user:
             raise GraphQLError("User not found")
         new_location = LocationModel.query.filter_by(
-          id=location_id, state="active").first()
+            id=location_id, state="active").first()
```
Comment (rajeman) [inline api/user/schema.py]: and this...

File: tests/test_user/test_create_user.py [hunk 3]
```diff
@@ -48,4 +48,6 @@ def test_user_creation_with_non_andela_email(self):
         query_response = self.client.execute(
             non_Andela_email_mutation,
             context_value={'session': db_session})
-        self.assertEqual(query_response, non_Andela_email_mutation_response)
+
+        response_expected = non_Andela_email_mutation_response
+        self.assertEqual(query_response, response_expected)
```
Comment (rajeman) [inline tests/test_user/test_create_user.py]: why is this change necessary?

File: tests/test_user_roles/test_create_user_role.py [hunk 4]
```diff
@@ -1,12 +1,17 @@
 from tests.base import BaseTestCase
 from fixtures.user_role.user_role_fixtures import (
-    user_role_mutation_query, user_role_mutation_response
+    user_role_mutation_query,
+    user_role_mutation_response,
+    user_mutation_query_duplicate,
```
Comment (rajeman) [inline tests/test_user_roles/test_create_user_role.py]: this name is not descriptive enough

File: tests/test_user_roles/test_create_user_role.py [hunk 5]
```diff
@@ -26,5 +31,15 @@ def test_user_role_creation(self):
             user_role_mutation_query,
             context_value={'session': db_session})
 
-        expected_responese = user_role_mutation_response
-        self.assertEqual(execute_query, expected_responese)
+        expected_response = user_role_mutation_response
+        self.assertEqual(execute_query, expected_response)
```
Comment (rajeman) [inline tests/test_user_roles/test_create_user_role.py]: why is this change necessary?

File: tests/test_user_roles/test_create_user_role.py [hunk 6]
```diff
@@ -1,12 +1,17 @@
...
+    user_mutation_query_duplicate,
+    user_role_mutation_duplicate_response
 )
+
 from helpers.database import db_session
 from api.user.models import User
 
 import sys
 import os
+
```
Comment (rajeman) [inline tests/test_user_roles/test_create_user_role.py]: why is this necessary?

File: tests/test_user_roles/test_create_user_role.py (hunk 5 above)
Comment (cop1fab) [reply tests/test_user_roles/test_create_user_role.py]: There was a typo instead of response it was written ```expected_responese```
